core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -331,6 +331,30 @@ private[spark] object Utils extends Logging {
   }
 
   /**
+   * A file name may contain some invalid url characters, such as " ". This method will convert the
+   * file name to a raw path accepted by `java.net.URI(String)`.
+   *
+   * Note: the file name must not contain "/" or "\"
+   */
+  def encodeFileNameToURIRawPath(fileName: String): String = {
```

Comment:
We don't need to encode `colon`. The purpose here is: assume we have a uri prefix: `spark://localhost:1234/` and a file name: `abc xyz`, and want to concat the prefix and the file name to `spark://localhost:1234/abc%20xyz` so that we can pass it to `java.net.URI(String)`.